vanlife-backend/src/main/java/org/beerbower/vanlife/controllers/LocationController.java

```diff
@@ -53,7 +57,10 @@ public List<Location> getLocations(
             return locationRepository.findByNameContainingIgnoreCase(name);
         }
 
-        if (type != null && minLat != null && maxLat != null && minLon != null && maxLon != null) {
+        LocationType type = locationTypeRepository.findById(typeId)
```

Comment:
I think that you want to move this down inside the `if (typeId != null) {` block otherwise you might be passing a null typeId.